types/index.d.ts

```diff
@@ -0,0 +1,6075 @@
+  NUMPAD3: string;
+  /** The numpad four key. */
+  NUMPAD4: string;
+  /** The numpad five key. */
+  NUMPAD5: string;
+  /** The numpad six key. */
+  NUMPAD6: string;
+  /** The numpad seven key. */
+  NUMPAD7: string;
+  /** The numpad eight key. */
+  NUMPAD8: string;
+  /** The numpad nine key. */
+  NUMPAD9: string;
+
+  /** The numpad multiply (*) key. */
+  MULTIPLY: string;
+  /** The numpad add (+) key. */
+  ADD: string;
+  /** The numpad separator (=) key. */
+  SEPARATOR: string;
+  /** The numpad subtract (-) key. */
+  SUBTRACT: string;
+  /** The numpad decimal (.) key. */
+  DECIMAL: string;
+  /** The numpad divide (/) key. */
+  DIVIDE: string;
+
+  /** The F1 key. */
+  F1: string;
+  /** The F2 key. */
+  F2: string;
+  /** The F3 key. */
+  F3: string;
+  /** The F4 key. */
+  F4: string;
+  /** The F5 key. */
+  F5: string;
+  /** The F6 key. */
+  F6: string;
+  /** The F7 key. */
+  F7: string;
+  /** The F8 key. */
+  F8: string;
+  /** The F9 key. */
+  F9: string;
+  /** The F10 key. */
+  F10: string;
+  /** The F11 key. */
+  F11: string;
+  /** The F12 key. */
+  F12: string;
+  /** The meta (Windows) key. */
+  META: string;
+  /** The command (⌘) key. */
+  COMMAND: string;
+}
+
+export type NightwatchPage = {
+  [name: string]: () => EnhancedPageObject<any, any, any>;
+} & {
+  [name: string]: NightwatchPage;
+};
+
+export interface NightwatchApiCommands {
+  readonly WEBDRIVER_ELEMENT_ID: string;
+  readonly browserName: string;
+  readonly platformName: string;
+  __isBrowserName(browser: string, alternateName: string): boolean;
+  __isPlatformName(platform: string): boolean;
+  isIOS(): boolean;
+  isAndroid(): boolean;
+  isMobile(): boolean;
+  isSafari(): boolean;
+  isChrome(): boolean;
+  isFirefox(): boolean;
+  isEdge(): boolean;
+  isInternetExplorer(): boolean;
+  isOpera(): boolean;
+}
+
+export interface NightwatchAPI
+  extends SharedCommands,
+  WebDriverProtocol,
+  NightwatchCustomCommands,
+  NightwatchApiCommands {
+  baseUrl: string;
+  assert: Assert;
+  actions(options?: { async?: boolean; bridge?: boolean }): Actions;
+  expect: Expect;
+  ensure: Ensure;
+  verify: Assert;
+
+  appium: AppiumCommands;
+
+  page: NightwatchPage & NightwatchCustomPageObjects;
+
+  /**
+   * SessionId of the session used by the Nightwatch api.
+   */
+  sessionId: string;
+
+  /**
+   * Override the sessionId used by Nightwatch client with another session id.
+   */
+  setSessionId(sessionId: string): this;
+
+  options: NightwatchTestOptions;
+
+  Keys: NightwatchKeys;
+
+  currentTest: NightwatchTestSuite;
+
+  globals: NightwatchGlobals;
+
+  launchUrl: string;
+  launch_url: string;
+}
+
+// tslint:disable-next-line:no-empty-interface
+export interface NightwatchCustomPageObjects { }
+
+export interface NightwatchBrowser
+  extends NightwatchAPI,
+  NightwatchComponentTestingCommands,
+  NightwatchCustomCommands { }
+
+export interface NightwatchComponentTestingCommands {
```

Comment:
After discussing with @beatfactor and @swrdfish we need to see what happens if this goes in and there is a clash? Does simply override or does it error? Can we see download numbers of the version that has been released and maybe remove it there and handle fall out?

@garg3133 and @vaibhavsingh97 can you look into this for me?

Reply:
No problem at all! Plugins can override the types, so we can easily deprecate it.